### DISCUSSION TITLE
AccessDenied when installing to AWS

### DISCUSSION THREAD

**Discussion by jensenbox** [artilleryio/artillery#1480] [Artillery]
Hi there,

I am installing on AWS from GitHub using OpenID Connect federation. I am able to control the role that is being used to install Artillery Pro but have run into an issue I cannot get past.

I was getting a cloudformation IAM issue and was able to get past that but now it is giving me an error withou presenting what exactly is causing the problem.

My question is: What permissions does my role need to be able to deploy Artillery Pro Cloudformation Stack?

```
##[debug]Evaluating condition for step: 'Run artillery deploy --region us-west-2 --license-key-file artillery-pro-license_key.txt'
##[debug]Evaluating: success()
##[debug]Evaluating success:
##[debug]=> true
##[debug]Result: true
##[debug]Starting: Run artillery deploy --region us-west-[2](https://github.com/Noviconnect/core/runs/6888215870?check_suite_focus=true#step:9:2) --license-key-file artillery-pro-license_key.txt
##[debug]Loading inputs
##[debug]Loading env
Run artillery deploy --region us-west-2 --license-key-file artillery-pro-license_key.txt
  artillery deploy --region us-west-2 --license-key-file artillery-pro-license_key.txt
  shell: /usr/bin/bash -e {0}
  env:
    AWS_DEFAULT_REGION: us-west-2
    AWS_REGION: us-west-2
    AWS_ACCESS_KEY_ID: ***
    AWS_SECRET_ACCESS_KEY: ***
    AWS_SESSION_TOKEN: ***
##[debug]/usr/bin/bash -e /home/runner/work/_temp/222d805d-d7d8-4096-a876-554c1a[3](https://github.com/Noviconnect/core/runs/6888215870?check_suite_focus=true#step:9:3)7[4](https://github.com/Noviconnect/core/runs/6888215870?check_suite_focus=true#step:9:4)1f[8](https://github.com/Noviconnect/core/runs/6888215870?check_suite_focus=true#step:9:8).sh
AWS account ID: ***
Detecting an existing installation...
Creating/updating Artillery Pro backend components...
Waiting for CloudFormation...
Error: Access Denied (AccessDenied)
Error: Process completed with exit code 1.
##[debug]Finishing: Run artillery deploy --region us-west-2 --license-key-file artillery-pro-license_key.txt
```

Thanks!
Christian

**Comment by dyllandry** [comment]
Unsure if this helps, but when I got a similar error (`AccessDenied: Access Denied Code: AccessDenied`), the problem was that I was trying to run artillery from a different region than the region specified in the policy that granted my user the permissions artillery needs to run. I was running artillery while specifying us-east-2 when the relevant policy had us-east-1 inside it.

**Comment by hassy** [comment]
hi @jensenbox 👋 Artillery Pro is no longer actively maintained. Artillery's main distribution includes support for load testing from AWS Fargate (and AWS Lambda) now.

https://www.artillery.io/docs/load-testing-at-scale/aws-fargate
